Meadowlark-js/packages/meadowlark-core/src/middleware/LogAnonymizedRequestMiddleware.ts

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { isDebugEnabled, Logger } from '@edfi/meadowlark-utilities';
+import { MiddlewareModel } from './MiddlewareModel';
+
+export async function anonymizeAndLogRequestBody({
+  frontendRequest,
+  frontendResponse,
+}: MiddlewareModel): Promise<MiddlewareModel> {
+  // if there is a response already posted, we are done
+  if (frontendResponse != null) return { frontendRequest, frontendResponse };
+
+  if (isDebugEnabled() && !frontendRequest.middleware.resourceInfo.isDescriptor) {
```

Comment:
The original idea is that descriptors _inside the body_ do not need to be anonymized. This code _is_ anonymizing them. I now think that's okay, because the client will get a 400 response that tells them the descriptor doesn't exist, and that response can be logged too.

For debugging purposes, it will still be handy to see what the payload structure is for a descriptor, therefore let's remove this second condition.
```suggestion
  if (isDebugEnabled()) {
```

Reply:
Ooh... here's a bigger challenge: what about nested documents?

```
POST http://localhost:3000/local/v3.3b/ed-fi/StudentInterventionAssociations
content-type: application/json
authorization: bearer {{authToken4}}

{ 
    "studentReference": {
        "studentUniqueId": "s0zf6d1123d3e"
    },
    "interventionReference": {
        "interventionIdentificationCode": "111",
        "educationOrganizationId": 123
    }
}
```

-->

```
2023-01-04 14:37:74 debug 3ffbd49e-2229-4f1e-aad2-e952c2e46f37 Anonymized request body: {"studentReference":null,"interventionReference":null}
```

Reply:
@stevenarnold-ed-fi I created a [pull request into your branch](https://github.com/Ed-Fi-Exchange-OSS/Meadowlark/pull/177), please take a look.